shuup/testing/utils.py

```diff
@@ -45,6 +49,18 @@ def apply_request_middleware(request, **attrs):
         activate(current_language)
 
     assert request.shop
+
+    if not attrs.get("skip_session", False):
+        frm = inspect.stack()[1]
+        mod = inspect.getmodule(frm[0])
+        if mod.__name__.startswith("shuup_tests.admin"):
+            # admin uses `get_shop` which relies on this "sessioned" shop id
+            if not request.session:
```

Comment:
Use `shuup.admin.set_providers.set_shop` here to ensure you are using same logic as the Shuup admin.